containers/src/Kokkos_DynRankView.hpp

```diff
@@ -1701,6 +1701,24 @@ auto as_view_of_rank_n(DynRankView<T, Args...> v) {
   return View<typename RankDataType<T, N>::type, Args...>(v.data(), v.layout());
 }
 
+template <typename Function, typename... Args>
+void apply_to_view_of_static_rank(Function&& f, DynRankView<Args...> a) {
+  switch (rank(a)) {
+    case 0: f(as_view_of_rank_n<0>(a)); break;
+    case 1: f(as_view_of_rank_n<1>(a)); break;
+    case 2: f(as_view_of_rank_n<2>(a)); break;
+    case 3: f(as_view_of_rank_n<3>(a)); break;
+    case 4: f(as_view_of_rank_n<4>(a)); break;
+    case 5: f(as_view_of_rank_n<5>(a)); break;
+    case 6: f(as_view_of_rank_n<6>(a)); break;
+    case 7: f(as_view_of_rank_n<7>(a)); break;
+    default:
+      Kokkos::Impl::throw_runtime_exception(
+          "Trying to apply a function to a view of unexpected rank " +
+          std::to_string(rank(a)));
+  }
+}
+
```

Comment:
If I could make this variadic in the `View`ish arguments, then it could be used for the non-scalar `deep_copy` implementations, too. It would then have to handle mixes of `View` and `DynRankView`. With C++17, that could just be a folded logical-or over the condition in the `enable_if`.